api/api/views/generate_teams.py

```diff
@@ -1,24 +1,48 @@
 from rest_framework import viewsets
 from rest_framework.decorators import action
+from rest_framework.request import Request
 from schema import SchemaError
 
+import api
 from api.ai.algorithm_runner import AlgorithmRunner
+from api.api.models import APISource
 from api.api.utils.generate_teams_data_loader import GenerateTeamsDataLoader
+from api.api.utils.logger import APILogger
 from api.api.utils.response_with_metadata import ResponseWithMetadata
 from api.api.validators.generate_teams_validator import GenerateTeamsValidator
 from api.dataclasses.team_set.serializer import TeamSetSerializer
 
 
 class GenerateTeamsViewSet(viewsets.GenericViewSet):
     @action(url_path="teams", detail=False, methods=["POST"])
-    def generate_teams(self, request):
+    def generate_teams(self, request: Request):
         try:
             request_data = dict(request.data)
 
             # validate data schema
             GenerateTeamsValidator(request_data).validate()
             input_data = GenerateTeamsDataLoader(request_data).load()
 
+            # Lowercase the params
+            params = {k.lower(): v for k, v in request.query_params.items()}
```

Comment:
are they not already lowercase?

Reply:
No, if you entered "?Sandbox=true" it comes through as "Sandbox", not "sandbox"